### ISSUE TITLE
LoRA layer param.grad=None after loss.backward() using lora for mamba 2.8B

### ISSUE THREAD

**Issue by jinlHe** [state-spaces/mamba#279]
I'm not sure which community I need to turn to for help, so I'm posting them all, I would appreciate it if someone could answer my questions:
[LoRA layer param.grad=None after loss.backward() using lora for mamba 2.8B](https://github.com/huggingface/peft/issues/1612#issue-2218459399)

**Comment by titoghose** [comment]
I'm facing the same issue while using trying to use LoRA to finetune. I have tried the peft library LoRA implementation, and also a simple LoRA wrapper class that I wrote myself. In both cases, the issue persists.

I think this is because if the `use_fast_path=True`, the forward pass of the `Mamba` module uses the `mamba_inner_fn`. This function requires only the `weight` variable of the linear layers, and does not use the `forward` method of `torch.nn.Linear`. Hence, even though the select linear layers of the model are wrapped using LoRA from the *peft* library, the LoRA matrices never accumulate gradients because I assume they are not added to the computational graph since the `mamba_inner_fn` only accesses their `weight` variable. 

**Partial solution**
If I set `use_fast_path=False`, the LoRA finetuning works just fine. However, in my case, this is not a possibility because with use_fast_path turned off, the GPU memory usage is too high and causes a CUDA Out of Memory Error. 


Still don't have a solution to making LoRA work (huggingface peft or my own implementation) with `use_fast_path=True`, so any help would be appreciated.
